libs/sdk-common/src/input_parser.rs

```diff
@@ -228,6 +249,68 @@ fn prepend_if_missing(prefix: &str, input: &str) -> String {
         false => format!("{}{}", prefix, input.trim_start_matches(prefix)),
     }
 }
+/// Parse a Liquid BIP21 String to retrieve [LiquidAddressData]
+pub fn parse_liquid_bip21(uri: &str) -> Result<AddressData> {
+    let mut parts = uri.split('?');
+
+    let Some(network_address) = parts.next() else {
+        return Err(anyhow!(
+            "Unable to extract network and address from bip21 string"
+        ));
+    };
+
+    // Extract network and address
+    let mut network_address_parts = network_address.split(':');
+    let network = match network_address_parts.next() {
+        Some("liquid") => Network::Bitcoin,
+        Some("liquidtestnet") => Network::Testnet,
+        Some(_) | None => {
+            return Err(anyhow!("Unable to extract network from bip21 string"));
+        }
+    };
+
+    let address = match network_address_parts.next() {
+        Some(address) => {
+            address.parse::<elements::Address>()?;
+            address.into()
+        }
+        None => {
+            return Err(anyhow!("Unable to extract address from bip21 string"));
+        }
+    };
+
+    if let Some(params) = parts.next() {
+        // Parse URI parameters
+        let mut pairs: HashMap<String, String> = HashMap::new();
+        for pair in params.split('&') {
+            let Some((key, val)) = pair.split_once('=') else {
+                continue;
+            };
+            pairs.insert(key.to_string(), val.to_string());
+        }
+
+        let amount_sat = match pairs.get("amount") {
+            Some(amount) => Some((amount.parse::<f64>()? * 100_000_000.0) as u64),
+            None => None,
+        };
+
+        return Ok(AddressData {
+            address,
+            network,
+            amount_sat,
+            label: pairs.get("label").cloned(),
+            message: pairs.get("message").cloned(),
+        });
+    }
+
+    Ok(AddressData {
+        address,
+        network,
+        amount_sat: None,
+        label: None,
+        message: None,
+    })
```

Comment:
NIT: One suggestion for readability.

We can define URI variables & assign their values directly after parsing, without the need for a `HashMap`.
```rust
    let mut amount_sat = None;
    let mut label = None;
    let mut message = None;
    
    if let Some(params) = parts.next() {
        for pair in params.split('&') {
            if let Some((key, val)) = pair.split_once('=') {
                match key {
                    "amount" => amount_sat = Some((val.parse::<f64>()? * 100_000_000.0) as u64),
                    "label" => label = Some(val.to_string()),
                    "message" => message = Some(val.to_string()),
                    _ => {}
                }
            }
        }
    }

    Ok(AddressData {
        address,
        network,
        amount_sat,
        label,
        message,
    })
```

One other improvement on earlier statements would be to use `ok_or`/`ok_or_else` & return error messages early, instead of pattern matching.

Reply:
Thanks for the suggestion Erdem, that looks very elegant!